nautobot/docs/development/apps/api/platform-features/prepopulating-data.md

```diff
@@ -0,0 +1,183 @@
+# Prepopulating Data
+
+In many cases, an app may wish to make use of Nautobot's various extensibility features, such as [custom fields](../../../../user-guide/platform-functionality/customfield.md) or [relationships](../../../../user-guide/platform-functionality/relationship.md). It can be useful for an app to automatically create a custom field definition or relationship definition as a consequence of being installed and activated, so that everyday usage of the app can rely upon these definitions to be present.
+
+Furthermore, sometimes apps might want to create a baseline of available data. This could, for example, be a baseline of circuit providers for a [Single Source of Truth](https://docs.nautobot.com/projects/ssot/en/latest/) integration that synchronizes circuit data or creating new default [statuses](https://docs.nautobot.com/projects/core/en/stable/user-guide/platform-functionality/status/) and/or [roles](https://docs.nautobot.com/projects/core/en/stable/user-guide/platform-functionality/role/), or adding to the allowed content types for existing statuses and roles.
+
+## Overview
+
+Multiple options for the population of base state are available, each with their own trade-offs. There is no single best option, you have to choose the one has the best properties for your specific need.
+
+- Signal Handlers on `nautobot_database_ready`
+- [Django Onboard Tooling](https://docs.djangoproject.com/en/4.2/howto/initial-data/)
+    - [Django Data Migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations)
+    - [Django Fixtures](https://docs.djangoproject.com/en/4.2/topics/db/fixtures/#fixtures)
+- [Design Builder Jobs](https://docs.nautobot.com/projects/design-builder/en/latest/)
+- Creating data in-place where needed
+
+Some of these methods are explained in further detail at the bottom of this page.
+
+The following table details points that are somewhat easily comparable between the different options.
+
+| Feature                        | Signal Handlers on `nautobot_database_ready`             | Django Data Migrations                      | Django Fixtures                                          | Design Builder Jobs                                      | Creating Data In-place Where Needed                      |
+|--------------------------------|----------------------------------------------------------|---------------------------------------------|----------------------------------------------------------|----------------------------------------------------------|----------------------------------------------------------|
+| Startup Impact                 | Exceptions occur after database migrations               | Exceptions occur during database migrations | N/A                                                      | N/A                                                      | N/A                                                      |
```

Comment:
Might suggest bolding the left column here (`**Startup Impact**`, `**Execution**`, etc.) to improve readability.